components/messages/MessageList.tsx

```diff
@@ -1,40 +1,65 @@
-import React, { useState, useRef } from 'react';
-import { ScrollView, StyleSheet } from 'react-native';
+import React, { useState, useRef, useEffect } from 'react';
+import { ScrollView, StyleSheet, Text } from 'react-native';
+import { useQuery } from '@apollo/client';
+import { GestureHandlerRootView } from 'react-native-gesture-handler';
 
-import { FakeMessage } from './Fakemessages';
+import { GET_CONTACT_MESSAGES } from '../../graphql/queries/Contact';
+import { COLORS } from '../../constants';
+import LoadingPage from '../ui/Loading';
 import Message from './Message';
 
-const MessagesList = ({ onSwipeToReply }: any) => {
-  const [messages, setMessages] = useState(FakeMessage);
-
-  const user = useRef(0);
+const MessagesList = ({ onSwipeToReply, userData }: any) => {
+  const [messages, setMessages] = useState([]);
   const scrollView = useRef();
 
-  return (
+  const variables = {
+    filter: { id: userData.id },
+    contactOpts: { limit: 1 },
+    messageOpts: { limit: 10, offset: 1 },
+  };
+  const { loading, error, data } = useQuery(GET_CONTACT_MESSAGES, { variables });
+
+  useEffect(() => {
+    if (error) {
+      console.log(error);
+    } else if (data) {
+      setMessages(data.search[0].messages);
+    }
+  }, [error, data]);
```

Comment:
Prefer to avoid using useEffect unless absolutely necessary, since this is difficult to understand when someone reads the code and try to check the flow of execution.

```suggestion
  let messages= []
  if(loading) return <Loading />
   if (error) {
      console.log(error);
    } 
     if (data) {
     messages = data.search[0].messages;
    }
```